tools/f5map/lib/f5map/parser.rb

```diff
@@ -0,0 +1,34 @@
+# Copyright 2018 Google LLC. This software is provided as-is,
+# without warranty or representation for any use or purpose.
+# Your use of it is subject to your agreements with Google.
+
+require 'treetop'
+require 'f5map/node_extensions'
+module F5map
+  ##
+  # The f5 configuration parser.
+  class Parser
+    # Load the grammar
+    Treetop.load(File.expand_path('../f5.treetop', __FILE__))
+    @@parser = F5PersistParser.new
+
+    def self.parse(data)
+      tree = @@parser.parse(data)
+      # self.clean_tree!(tree)
+      tree
+    end
+
+    def self.parser
+      @@parser
+    end
+
+    private
+
+    # Would be nice to make this functional instead of imperative.
+    def self.clean_tree!(root_node)
```

Comment:
This method is not called.

Reply:
Will look into this...  That strikes me as strange.  I may have committed the code wired for debugging.

Reply:
For reference, line 17 shows the only call to this method commented out.